src/test/java/uk/gov/hmcts/dts/fact/services/MapitServiceTest.java

```diff
@@ -124,4 +129,42 @@ void shouldReturnOptionalEmptyIfBlankPostcodeForPartial() {
         assertThat(result).isNotPresent();
         verifyNoInteractions(mapitClient);
     }
+
+    @Test
+    void localAuthorityExistsShouldReturnTrueForValidLocalAuthorityNames() {
+        LinkedHashMap<String, MapitArea> mockAreaData = new LinkedHashMap<>();
+        mockAreaData.put("100", new MapitArea("100", "Birmingham City Council", "MTD"));
+        when(mapitClient.getMapitDataForLocalAuthorities(any())).thenReturn(mockAreaData);
+
+        assertThat(mapitService.localAuthorityExists("Birmingham City Council")).isTrue();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "", " " })
+    @NullSource
+    void localAuthorityExistsShouldReturnFalseForEmptyLocalAuthorityNames(String name) {
+        assertThat(mapitService.localAuthorityExists(name)).isFalse();
+
+        // We don't need to call the client to validate the name if the name is
+        // null, empty or whitespace
+        verify(mapitClient, never()).getMapitDataForLocalAuthorities(any());
+    }
+
+    @Test
+    void localAuthorityExistsShouldReturnFalseForInvalidLocalAuthorityName() {
+        when(mapitClient.getMapitDataForLocalAuthorities(any())).thenReturn(new LinkedHashMap<>());
```

Comment:
Again nothing wrong with this. An alternative to creating an empty map is simply use emptyMap() from java.util.Collections class

Reply:
Good suggestion. I've changed to use the alternative emptyMap method.